packages/lesswrong/lib/collections/posts/constants.ts

```diff
@@ -31,3 +35,16 @@ export const postStatusLabels = [
     label: 'deleted'
   }
 ];
+
+const amaTagIdSetting = new DatabasePublicSetting<string | null>('amaTagId', null)
+const openThreadTagIdSetting = new DatabasePublicSetting<string | null>('openThreadTagId', null)
+const startHerePostIdSetting = new DatabasePublicSetting<string | null>('startHerePostId', null)
+
+export const postTagIcons = new Map([
+  [amaTagIdSetting.get(), QuestionAnswerIcon], 
```

Comment:
Ah, here you hit a problem. You're trying to get a database setting (which is asynchronously loaded after initialization), at initialization. This is causing your tests to fail.

If you manage to pull off my suggested refactor, this problem should be obviated, but otherwise, you'll need to do some indirection to make sure you don't read database settings until they're ready.

Reply:
Thanks, fixed in 73a920146455cfc79aaff981d25d9ecd7fbcb8fe.